cupy/fallback_mode/fallback.py

```diff
@@ -0,0 +1,139 @@
+"""
+Main fallback class.
+"""
+import types
+import cupy as cp
+import numpy as np
+
+
+from cupy.fallback_mode.utils import _call_cupy
+from cupy.fallback_mode.utils import _call_numpy
+
+
+class _RecursiveAttr:
+    """
+    RecursiveAttr class to catch all attributes corresponding to numpy,
+    when user calls fallback_mode. numpy is an instance of this class.
+    """
+
+    to_notify = True
+
+    @classmethod
+    def notifications(cls):
+        """
+        Return notification status
+        """
+        return cls.to_notify
+
+    @classmethod
+    def set_notifications(cls, set_to):
+        """
+        Set notification status to bool set_to.
+        """
+        cls.to_notify = set_to
+
+        if cls.to_notify:
+            print("Notifications are Enabled")
+        else:
+            print("Notifications are Disabled")
+
+    def __init__(self, numpy_object, cupy_object):
+
+        self._numpy_object = numpy_object
+        self._cupy_object = cupy_object
+
+    @property
+    def _cupy_module(self):
+        if isinstance(self._cupy_object, types.ModuleType):
+            return self._cupy_object
+        raise TypeError("'{}' is not a module"
+                        .format(self._cupy_object.__name__))
+
+    @property
+    def _numpy_module(self):
+        if isinstance(self._numpy_object, types.ModuleType):
+            return self._numpy_object
+        raise TypeError("'{}' is not a module"
+                        .format(self._numpy_object.__name__))
+
+    def __getattr__(self, attr):
+        """
+        Catches attributes corresponding to numpy.
+
+        Runs recursively till attribute gets called.
+        Or numpy ScalarType is retrieved.
+
+        Args:
+            attr (str): Attribute of _RecursiveAttr class object.
+
+        Returns:
+            (_RecursiveAttr object, NumPy scalar):
+            Returns_RecursiveAttr object with new numpy_object, cupy_object.
+            Returns module, scalars if requested.
+        """
+        # getting attr
+        numpy_object = getattr(self._numpy_object, attr, None)
+        cupy_object = getattr(self._cupy_object, attr, None)
+
+        # Retrieval of NumPy scalars
+        if isinstance(numpy_object, np.ScalarType):
+            return numpy_object
+
+        return _RecursiveAttr(numpy_object, cupy_object)
+
+    def __repr__(self):
+
```

Comment:
It's okay, but `__repr__` of `numpy.modules` will be quite messy.
```
>>> numpy.linalg
<numpy = <module 'numpy.linalg' from 'C:\\Users\\Piyush-Laptop\\AppData\\Local\\conda\\conda\\envs\\cupy_dev\\lib\\site-packages\\numpy\\linalg\\__init__.py'>>, cupy = <module 'cupy.linalg' from 'c:\\users\\piyush-laptop\\desktop\\gsoc\\cupy\\cupy\\linalg\\__init__.py'>
```

Reply:
I think it is better to show both numpy and cupy object in any cases. How do you think?

Reply:
Okay I'll show both. For modules:
`<numpy=module {self._numpy_object.__name__}, cupy=module {self._cupy_object.__name__}>`
This will make it less messy. Is that okay?

Reply:
Looks good!

Reply:
Sorry, None` doesn't have `__name__` attribute, and sometimes `self._{numpy/cupy}_object` is not a module.

Reply:
I mean something like this:
```
def __repr__(self):
    if isinstance(self._numpy_object, types.ModuleType):
        return f"<numpy = module {self._numpy_object.__name__}, cupy = module {getattr(self._cupy_object, '__name__', None)}>"
    return f"<numpy = {self._numpy_object}, cupy = {self._cupy_object}>"
```